lte/gateway/c/oai/tasks/amf/amf_app_main.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "log.h"
+#include "intertask_interface_types.h"
+#include "intertask_interface.h"
+#include "itti_free_defined_msg.h"
+#include "service303_message_utils.h"
+#include "amf_as_message.h"
+#ifdef __cplusplus
+}
+#endif
+#include "amf_app_messages_types.h"
+#include "amf_config.h"
+#include "amf_data.h"
+#include "amf_fsm.h"
+#include "amf_app_ue_context_and_proc.h"
+#include "amf_app_defs.h"
+#include "ngap_messages_types.h"
+#include "amf_app_state_manager.h"
+#include "common_defs.h"
+using namespace std;
+namespace magma5g {
+task_zmq_ctx_t amf_app_task_zmq_ctx;
+void amf_app_exit(void);
+static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
+  zframe_t* msg_frame = zframe_recv(reader);
+  assert(msg_frame);
+  MessageDef* received_message_p = (MessageDef*) zframe_data(msg_frame);
+  imsi64_t imsi64                = itti_get_associated_imsi(received_message_p);
+  amf_app_desc_t* amf_app_desc_p = get_amf_nas_state(false);
+
+  switch (ITTI_MSG_ID(received_message_p)) {
+    case NGAP_INITIAL_UE_MESSAGE:
+      OAILOG_DEBUG(LOG_AMF_APP, "NGAP_INITIAL_UE_MESSAGE received\n");
+      amf_app_handle_initial_ue_message(
+          amf_app_desc_p, &NGAP_INITIAL_UE_MESSAGE(received_message_p));
+      break;
+    case AMF_APP_UPLINK_DATA_IND:
+      OAILOG_DEBUG(LOG_AMF_APP, "UPLINK_NAS_MESSAGE received\n");
+      amf_app_handle_uplink_nas_message(
+          amf_app_desc_p, AMF_APP_UL_DATA_IND(received_message_p).nas_msg);
+      break;
+    case N11_CREATE_PDU_SESSION_RESPONSE:
+      OAILOG_DEBUG(
+          LOG_AMF_APP,
+          "session created in SMF for imsi:%s for session_id:%d \n",
+          N11_CREATE_PDU_SESSION_RESPONSE(received_message_p).imsi,
+          N11_CREATE_PDU_SESSION_RESPONSE(received_message_p).pdu_session_id);
+      amf_app_handle_pdu_session_response(
+          &N11_CREATE_PDU_SESSION_RESPONSE(received_message_p));
+      break;
+    case NGAP_PDUSESSIONRESOURCE_SETUP_RSP:
```

Comment:
@sanjay-ACL @sureshgorijavolu can we have a consistent naming convention here? This can be fixed as part of a separate PR.

```suggestion
    case NGAP_PDU_SESSION_RESOURCE_SETUP_RSP:
```

Reply:
will fix this in separate PR